middlewares/validations/validations.js

```diff
@@ -0,0 +1,47 @@
+class Validations {
+  static async validateCreateUser(req, res, next) {
+    const { username, email, password } = req.body;
+    const emailRegex = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/;
+    const pwdRegex = /^(?=.*?[A-Z])(?=(.*[a-z]){1,})(?=(.*[\d]){1,})(?=(.*[\W]){1,})(?!.*\s).{8,}$/;
+
+    switch (true) {
+      case email == null || email === undefined || password == null || password === undefined
```

Comment:
Implementation of `==` is in a wrong use, since `==` checks for the values of the operands not there types, forexample `77 == '77'` will evalute to `true` yet one of the operands is a `Number` and the other a `String` which should actually evaluate to `false`. Since in `javascript`, `null` is an object the comparison you have implemented in many situations it will break, kindly change `==` to the triple comparison operator `===`